iree/task/task.c

```diff
@@ -287,7 +287,15 @@ static uint32_t iree_math_ptr_to_xrgb(const uintptr_t ptr) {
   // This is just a simple hack to give us a unique(ish) per-pointer color.
   // It's only to make it easier to distinguish which tiles are from the same
   // dispatch.
-  return (uint32_t)ptr ^ (uint32_t)(ptr >> 32);
+  #if UINTPTR_MAX == 0xFFFF
+    return (uint32_t) ptr;
+  #elif UINTPTR_MAX == 0xFFFFFFFF
+    return (uint32_t) ptr;
+  #elif UINTPTR_MAX == 0xFFFFFFFFFFFFFFFFu
```

Comment:
ah, if doing that then use the IREE_PTR_SIZE_32 / IREE_PTR_SIZE_64 macros (those are the only pointer sizes we will ever support)
the 0xFFFFFFFFFFFFFFFFFFFFu approach won't always work as it'd need to be ull (as u is 32-bit, so it could get truncated to 32-bit and never be equal to UINTPTR_MAX)

Reply:
done.